cloudregister/registerutils.py

```diff
@@ -346,30 +346,51 @@ def enable_repository(repo_name):
     """Enable the given repository"""
 
     cmd = ['zypper', 'mr', '-e', repo_name]
-    res = exec_subprocess(cmd)
-    if res:
-        log.error('Unable to enable repository %s' % repo_name)
+    _, _, failed = exec_subprocess(cmd)
+    if failed:
+        log.error(
+            'Unable to enable repository {}'.format(repo_name)
+        )
 
 
 # ----------------------------------------------------------------------------
-def exec_subprocess(cmd, return_output=False):
-    """Execute the given command as a subprocess (blocking)
-       Returns one off:
-           - exit code of the command
-           - stdout, stderr and exit code
-           - -1 indicates an exception"""
+def exec_subprocess(cmd, tolog=True, disconnect=False):
+    """
+    Execute the given command as a subprocess (blocking)
+    Returns a tuple list:
+        (binary_stdout, binary_stderr, exitcode)
+        A -1 exitcode indicates an OSError at command invocation
+    """
+    stdout = b''
+    stderr = b''
+    ecode = -1
```

Comment:
`ecode` may imply "error code" that in turns implies that we expect an error. But we shouldn't so maybe this should be called `rcode`

Reply:
done